### PR TITLE
#933 Feature/imagemagick cli adapter

### PR DESCRIPTION
**Pull request by mrudnicki**

### CODE REVIEW

File: docs/Development_Documentation/04_Assets/03_Working_with_Thumbnails/01_Image_Thumbnails.md [hunk 1]
```diff
@@ -47,6 +46,26 @@ This path can then be directly used to display the image in a `<img />` or `<pic
...
+If `convert` and `composite` are accessible globally:
+
+```php
+\Pimcore\Image\Adapter::class => DI\object('Pimcore\Image\Adapter\ImageMagick')
+```
+
+You can also specify the path to tools, like below:
+```php
+\Pimcore\Image\Adapter::class => DI\object('Pimcore\Image\Adapter\ImageMagick')
+        ->method('setConvertScriptPath', '/usr/bin/convert')
```
Comment (brusch) [inline docs/Development_Documentation/04_Assets/03_Working_with_Thumbnails/01_Image_Thumbnails.md]: I think this configurations should be optional. 
Pimcore has an locator for binaries, which should be used in `getConvertScriptPath()` and `getCompositeCommand()` if the path was not set manually before. 

This is how to use the locator: 
```php
$path = \Pimcore\Tool\Console::getExecutable("convert");
$path = \Pimcore\Tool\Console::getExecutable("composite");
```

File: pimcore/lib/Pimcore/Image/Adapter/ImageMagick.php [hunk 2]
```diff
@@ -0,0 +1,772 @@
...
+    }
+
+    /**
+     * Returns the composite cli script path.
+     *
+     * @return string
+     */
+    public function getCompositeScriptPath()
+    {
+        return $this->compositeScriptPath;
```
Comment (brusch) [inline pimcore/lib/Pimcore/Image/Adapter/ImageMagick.php]: Locator should be used here.

Comment (brusch) [inline pimcore/lib/Pimcore/Image/Adapter/ImageMagick.php]: Locator should be used here.

Comment (mrudnicki) [inline pimcore/lib/Pimcore/Image/Adapter/ImageMagick.php]: Good I change it

**Pull request merged**